src/allowance/utils.ts

```diff
@@ -29,15 +29,30 @@ export const getApproved = async (
   }
 }
 
-export const setApproval = (
+export const setApproval = async (
   signer: Signer,
   tokenAddress: string,
   contractAddress: string,
   amount: string
 ): Promise<ContractTransaction> => {
   const erc20 = new Contract(tokenAddress, ERC20_ABI, signer) as ERC20Contract
+  const transactionRequest = await erc20.populateTransaction.approve(
+    contractAddress,
+    amount
+  )
+
+  try {
+    const estimatedGasLimit = await signer.estimateGas(transactionRequest)
+    if (estimatedGasLimit) {
+      const formattedGasLimit = ethers.utils.parseEther(
+        `${(BigInt(estimatedGasLimit.toString()) * 125n) / 100n}`
+      )
+
+      transactionRequest.gasLimit = formattedGasLimit
```

Comment:
Are we sure about bumping the gas limit by 25% right away?

Reply:
In other parts, we bump it if there is an error with current estimation.

Reply:
This change was done based on an internal discussion. Sharing the discord link in DM.